client/src/components/Onboarding/OnboardingActions.js

```diff
@@ -0,0 +1,114 @@
+import React, { useCallback } from 'react';
+import { useRouteMatch } from 'react-router-dom';
+import { useSelector, useDispatch } from 'react-redux';
+import { LoadingSpinner, Message } from '_common';
+import { Button } from 'reactstrap';
+import { stepPropType } from './OnboardingPropTypes';
+import './OnboardingActions.module.scss';
+
+const OnboardingActions = ({ step }) => {
+  const dispatch = useDispatch();
+  const actionCallback = useCallback((action, username) => {
+    dispatch({
+      type: 'POST_ONBOARDING_ACTION',
+      payload: {
+        step: step.step,
+        action,
+        username
+      }
+    });
+  });
+  const isStaff = useSelector(state =>
+    state.authenticatedUser.user ? state.authenticatedUser.user.isStaff : false
+  );
+  const isSending = useSelector(
+    state =>
+      state.onboarding.action.loading &&
+      state.onboarding.action.step === step.step
+  );
+  const error = useSelector(state => state.onboarding.action.error);
+  const actionStep = useSelector(state => state.onboarding.action.step);
+  const { params } = useRouteMatch();
+  const authUsername = useSelector(state =>
+    state.authenticatedUser.user ? state.authenticatedUser.user.username : ''
+  );
+
+  const hasSendingError = actionStep === step.step && error;
+
+  // If the route loaded shows we are viewing a different user
+  // (such as an admin viewing a user) then pull the username for
+  // actions from the route. Otherwise, use the username of whomever is logged in
+  const username = params.username || authUsername;
+
+  if (hasSendingError) {
+    return (
+      <Message type="warn" className="appDetail-error">
+        We were unable to perform this action.
+      </Message>
+    );
+  }
+
+  return (
+    <span styleName="root">
+      {isStaff && step.state === 'staffwait' ? (
+        <>
+          <Button
+            color="link"
+            styleName="action"
+            disabled={isSending}
+            onClick={() => actionCallback('staff_approve', username)}
+          >
+            <h6>{step.staffApprove}</h6>
```

Comment:
__Notice__ _(no change requested, knowledge sharing)_

An `<h6>` within a `<button>` is "not allowed" ([HTML validator check](https://validator.w3.org/nu/#textarea), MDN*). I know this is likely copied from other locations (where this must be fixed, also), so I get how it's happened. The rules of HTML element nesting became more complex, last decade; but a cheatsheet rule-of-thumb is to use tags to describe content, not conveniently style content (i.e. don't rely on bold and font-size from `<h6>` for `<button>`).

* A `<button>` should _only_ have "phrasing content". An `<h1>`–`<h6>` is _not_ "phrasing content". Also, an `<h1>`–`<h6>`'s "Permitted parents" are "Any element that accepts flow content"; a `<button>`'s "Permitted content" is _not_ flow content. _(source: [`<button>` | MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button), see table beneath demo; [`<h1>`–`<h6>` | MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements), see table beneath demo)_

Reply:
Removing the h6 makes those action links not bold. Is there a pre-existing style that I can apply?

Reply:
Nope, but you have a `.action` class; so you can add `font-weight: 500` to that.

> There are shared styles for buttons (meant to be abstracted by the incomplete #100). @rstijerina has been contacting me, trying to use those directly (not intended, but can be done). I hope to help after scrum. _Sample usage: `DataFilesSearchbar.module.css` [line #33](https://github.com/TACC/Frontera-Portal/blob/master/client/src/components/DataFiles/DataFilesSearchbar/DataFilesSearchbar.module.css#L33), [line #51](https://github.com/TACC/Frontera-Portal/blob/master/client/src/components/DataFiles/DataFilesSearchbar/DataFilesSearchbar.module.css#L51), [line #54](https://github.com/TACC/Frontera-Portal/blob/master/client/src/components/DataFiles/DataFilesSearchbar/DataFilesSearchbar.module.css#L54)_